### ISSUE TITLE
Intercepting responses causes multiple Set-Cookie headers to be folded into just one Set-Cookie header

### ISSUE THREAD

**Issue by cheez3d** [wkeeling/selenium-wire#314]
Hello,

I am using **selenium-wire** for intercepting and modifying responses. One such response corresponds to a login request on a website. Consequently, the server will add a couple of (in this case 3) `Set-Cookie` headers to its response, for keeping track of the authenticated user.

The issue arises from [this line](https://github.com/wkeeling/selenium-wire/blob/master/seleniumwire/handler.py#L144) or [this line](https://github.com/wkeeling/selenium-wire/blob/master/seleniumwire/mitmproxy.py#L152) in **selenium-wire** (depending on whether **mitmproxy** is set as the back-end or not), specifically the `items()` call, which by default will have the argument `multi=False` which causes all `Set-Cookie` headers to be folded into just one `Set-Cookie` header, with all cookies separated by commas. This makes manipulating different cookies separately harder, because you have to split the folded cookies into separate cookies (which is a pain to get right).

What actually made me discover this issue was the fact that, when enabling the interceptor for intercepting the login response, the login mechanism of the website broke (after logging in, I would still be logged out). I have narrowed it down to this cookie folding behavior of `items()`. It seems that Chrome only stores the first cookie from a `Set-Cookie` header with multiple cookies, ignoring the rest, at least in my scenario.

Here is a concrete example of how the headers are sent by the server, and how they end up being received by the browser when the interceptor is enabled:

#### Sent by the server
```http
Set-Cookie: xf_user=34617,Y_k18gWj8XhDARWjCj_XYVIokkOzvfc5MVkJWuSt; Domain=.something.com; Max-Age=1728000; Path=/; expires=Sun, 27-Jun-2021 17:30:45 GMT; secure; HttpOnly
Set-Cookie: xf_session=; Domain=.something.com; Max-Age=0; Path=/; expires=Wed, 02-Jun-2021 17:30:45 GMT
Set-Cookie: auth="eyJfdXNlciI6WzU2Nzk5NTAzNTg5MDQ4MzIsMSwiOFNicUlwTXc5YnFmb3E5WE1HTEtKZyIsMTYyMzA4NzA0NSwxNjIzMDg3MDQ1XX0\075|1623087046|51d18fe02628f468460bdca5e0010c2c66c78ea8"; Domain=.something.com; Max-Age=1814400; Path=/; expires=Mon, 28-Jun-2021 17:30:46 GMT
```

#### Received by browser
```http
Set-Cookie: xf_user=34617,Y_k18gWj8XhDARWjCj_XYVIokkOzvfc5MVkJWuSt; Domain=.something.com; Max-Age=1728000; Path=/; expires=Sun, 27-Jun-2021 17:30:45 GMT; secure; HttpOnly, xf_session=; Domain=.something.com; Max-Age=0; Path=/; expires=Wed, 02-Jun-2021 17:30:45 GMT, auth="eyJfdXNlciI6WzU2Nzk5NTAzNTg5MDQ4MzIsMSwiOFNicUlwTXc5YnFmb3E5WE1HTEtKZyIsMTYyMzA4NzA0NSwxNjIzMDg3MDQ1XX0\075|1623087046|51d18fe02628f468460bdca5e0010c2c66c78ea8"; Domain=.something.com; Max-Age=1814400; Path=/; expires=Mon, 28-Jun-2021 17:30:46 GMT
```

In this case, only the `xf_user` cookie will be stored by the browser and sent back to the server on future requests.

I was able to fix the broken login by changing `items()` to `items(multi=True)`, in order to keep the `Set-Cookie` headers separate when they are propagated to the browser after running the interceptor. The same fix should also work for request interceptors, where the same `items()` call is present.

Would this simple `multi=True` be a proper fix for this issue?

**Comment by wkeeling** [comment]
Thank you for raising such a clear and detailed issue. I think you're absolutely right - we should be preserving the separate headers with `multi=True` rather than implicitly combining them. Selenium Wire should be transparent as far as the user is concerned and definitely shouldn't be restructuring headers.

I'll update the code and tests with `multi=True` or if you wanted to put a PR in feel free. I've been a little busy recently but hopefully I'll get to it this week.

**Comment by wkeeling** [comment]
Fixed in 4.3.1
